### ISSUE TITLE
nxlog_source and nxlog_destination providers don't work for wrapper cookbooks

### ISSUE THREAD

**Issue by jeremiahsnapp** [widgit-cookbooks/nxlog#6]
The following linked lines of code cause the template resources in nxlog_source and nxlog_destination providers to look for the template's source file in the wrapper cookbook that has defined the nxlog_source and nxlog_destination resources.

https://github.com/widgit-cookbooks/nxlog/blob/v0.5.0/providers/source.rb#L132

https://github.com/widgit-cookbooks/nxlog/blob/v0.5.0/providers/destination.rb#L126

https://github.com/widgit-cookbooks/nxlog/blob/v0.5.0/providers/destination.rb#L134

For instance, I can put the following example from the nxlog README into the default recipe of a cookbook named "test-nxlog".

```
include_recipe 'nxlog::default'

nxlog_destination 'windows_events_file' do
  file 'c:/windows/temp/events.log'
end

nxlog_source 'event_log' do
  input_module 'im_msvistalog'
  destination 'windows_events_file'
end
```

When I run this with Chef 12 I get the following error.

```
PS C:\vagrant> chef-client -z -r test-nxlog
[2015-06-30T10:49:08-07:00] WARN: No config file found or specified on command line, using command line options.
Starting Chef Client, version 12.4.0
resolving cookbooks for run list: ["test-nxlog"]
Synchronizing Cookbooks:
  - test-nxlog
  - nxlog
Compiling Cookbooks...
Converging 7 resources
Recipe: nxlog::default
  * remote_file[nxlog] action create[2015-06-30T10:49:27-07:00] WARN: Mode 644 includes bits for the owner, but owner is not specified
[2015-06-30T10:49:27-07:00] WARN: Mode 644 includes bits for the group, but group is not specified
 (up to date)
  * windows_package[nxlog] action install (up to date)
  * windows_service[nxlog] action enable (up to date)
  * windows_service[nxlog] action start (up to date)
  * template[c:/Program Files (x86)/nxlog/conf/nxlog.conf] action create (up to date)
  * directory[c:/Program Files (x86)/nxlog/conf/nxlog.conf.d] action create (up to date)
Recipe: test-nxlog::default
  * nxlog_destination[windows_events_file] action create
    - Create nxlog_destination[windows_events_file]
  * template[c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/10_op_windows_events_file.conf] action create (up to date)
  * template[c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/op_windows_events_file.default] action create (skipped due to only_if)
  * nxlog_source[event_log] action create
    - Create nxlog_source[event_log]
  * template[c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/20_ip_event_log.conf] action create

    ================================================================================
    Error executing action `create` on resource 'template[c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/20_ip_event_log.conf]'
    ================================================================================

    Chef::Exceptions::FileNotFound
    ------------------------------
    Cookbook 'test-nxlog' (0.0.0) does not contain a file at any of these locations:
      templates/windows-6.3.9600/resources/source.conf.erb
      templates/windows/resources/source.conf.erb
      templates/default/resources/source.conf.erb
      templates/resources/source.conf.erb

    Resource Declaration:
    ---------------------
    # In C:/Users/vagrant/.chef/local-mode-cache/cache/cookbooks/nxlog/providers/source.rb

    131:     template config_filename(n.name) do
    132:       cookbook n.cookbook_name.to_s
    133:       source 'resources/source.conf.erb'
    134:
    135:       variables name: n.name,
    136:                 params: params,
    137:                 destinations: destinations
    138:
    139:       notifies :restart, 'service[nxlog]', :delayed
    140:     end
    141:   end

    Compiled Resource:
    ------------------
    # Declared in C:/Users/vagrant/.chef/local-mode-cache/cache/cookbooks/nxlog/providers/source.rb:131:in `block (2 levels) in class_from_file'

    template("c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/20_ip_event_log.conf") do
      action :create
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      path "c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/20_ip_event_log.conf"
      backup 5
      atomic_update true
      source "resources/source.conf.erb"
      cookbook "test-nxlog"
      variables {:name=>"event_log", :params=>[["Module", "im_msvistalog"]], :destinations=>["windows_events_file"]}
      declared_type :template
      cookbook_name "test-nxlog"
    end


Running handlers:
[2015-06-30T10:49:28-07:00] ERROR: Running exception handlers
Running handlers complete
[2015-06-30T10:49:28-07:00] ERROR: Exception handlers complete
Chef Client failed. 2 resources updated in 19.045975 seconds
[2015-06-30T10:49:28-07:00] FATAL: Stacktrace dumped to C:/Users/vagrant/.chef/local-mode-cache/cache/chef-stacktrace.out
[2015-06-30T10:49:28-07:00] FATAL: Chef::Exceptions::FileNotFound: template[c:/Program Files (x86)/nxlog/conf/nxlog.conf.d/20_ip_event_log.conf] (C:/Users/vagrant/.chef/local-mode-cache/cache/cookbooks/nxlog/providers/source.rb line 131) had an error: Chef::Exceptions::FileNotFound: Cookbook 'test-nxlog' (0.0.0) does not contain a file at any of these locations:
  templates/windows-6.3.9600/resources/source.conf.erb
  templates/windows/resources/source.conf.erb
  templates/default/resources/source.conf.erb
  templates/resources/source.conf.erb
```

If this behavior is intentional and you expect the wrapper cookbook test-nxlog to have copies of the template source files then this should be documented in nxlog's README.

If this behavior is not intentional then the lines of code linked above needs to be fixed. There are a couple of options.

One option is to set the values to `"nxlog"` so the nxlog's template source files will always be used.

**Comment by gh2k** [comment]
I didn't realise that would autoclose the issue. I'm still waiting for my teamcity build, and to run some more tests.

**Comment by gh2k** [comment]
Thanks for your thorough bug report. I've released this on Supermarket as version 0.6.0. Apologies for the error.

**Comment by jeremiahsnapp** [comment]
Awesome! Thanks for the quick fix @gh2k
